sdk/core/Microsoft.Azure.Core.NewtonsoftJson/src/NewtonsoftJsonObjectSerializer.cs

```diff
@@ -102,7 +102,7 @@ public override ValueTask SerializeAsync(Stream stream, object? value, Type inpu
 
             return _cache.GetOrAdd(member, m =>
             {
-                if (_serializer.ContractResolver.ResolveContract(m.ReflectedType) is JsonObjectContract contract)
+                if (m.ReflectedType != null && _serializer.ContractResolver.ResolveContract(m.ReflectedType) is JsonObjectContract contract)
```

Comment:
What happens if `m.ReflectedType` is null?  Will it skip this block?  Is that the desired behavior?

Reply:
`m.ReflectedType` would only be null in the case: 
> If the MemberInfo object is a global member (that is, if it was obtained from the [Module.GetMethods](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.module.getmethods?view=net-8.0) method, which returns global methods on a module), the returned [DeclaringType](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.memberinfo.declaringtype?view=net-8.0) will be null.

according to the [API remarks](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.memberinfo.reflectedtype?view=net-8.0#remarks). 

This is true for all versions of this API. The docs for `NewtonSoft.Json.Serialization` do not specify what happens if the type argument is null. From experimentation, it throws an argument null exception, but since it's not documented I guess it's technically an implementation detail. In this case it seems like it would make more sense to use null suppression to preserve current behavior.